### PR TITLE
gsim and glim functionalities added

### PR DESCRIPTION
**Pull request by nikpawar89**
GLIM and GSIM functionalities

### CODE REVIEW

File: fineract-provider/src/main/resources/sql/migrations/core_db/V344__GSIM_migration_script.sql [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	`parent_deposit` DECIMAL(19,6) NOT NULL DEFAULT '0.000000',
+	`child_accounts_count` INT(11) NOT NULL,
+	`accepting_child` TINYINT(4) NOT NULL DEFAULT '0',
+	`savings_status_id` SMALLINT(5) NOT NULL DEFAULT '0',
+	`application_id` DECIMAL(10,0) NULL DEFAULT '0',
+	PRIMARY KEY (`id`),
+	UNIQUE INDEX `gsim_account_no_UNIQUE` (`account_number`),
+	INDEX `FK_gsim_group_id` (`group_id`),
+	CONSTRAINT `FK_gsim_group_id` FOREIGN KEY (`group_id`) REFERENCES `m_group` (`id`)
+) COLLATE='utf8_general_ci' ENGINE=InnoDB AUTO_INCREMENT=1;
```
Comment (avikganguly01) [inline fineract-provider/src/main/resources/sql/migrations/core_db/V344__GSIM_migration_script.sql:55]: Please remove collate, engine and auto increment specifications from the migration scripts.

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/service/SavingsApplicationProcessWritePlatformServiceJpaRepositoryImpl.java [hunk 2]
```diff
@@ -74,472 +92,713 @@
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+
 @Service
```
Comment (avikganguly01) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/service/SavingsApplicationProcessWritePlatformServiceJpaRepositoryImpl.java:98]: Unreviewable code. Please make sure file diff for existing files is readable by changing only parts which you intend to change.

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/service/SavingsAccountWritePlatformServiceJpaRepositoryImpl.java [hunk 3]
```diff
@@ -250,9 +359,23 @@ public CommandProcessingResult deposit(final Long savingsId, final JsonCommand c
...
         boolean isAccountTransfer = false;
         boolean isRegularTransaction = true;
+
         final SavingsAccountTransaction deposit = this.savingsAccountDomainService.handleDeposit(account, fmt, transactionDate,
                 transactionAmount, paymentDetail, isAccountTransfer, isRegularTransaction);
 
+        	if(isGsim && (deposit.getId()!=null))
+        	{
+        		GroupSavingsIndividualMonitoring gsim=gsimRepository.findOne(account.getGsim().getId());
+        		System.out.println("parent deposit "+gsim.getParentDeposit());
```
Comment (avikganguly01) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/service/SavingsAccountWritePlatformServiceJpaRepositoryImpl.java:369]: Move to slf4j logger debug level if you wish to log something for debugging purposes.

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/DepositsApiConstants.java [hunk 4]
```diff
@@ -26,270 +26,276 @@
 
 public class DepositsApiConstants {
```
Comment (avikganguly01) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/DepositsApiConstants.java:28]: Unreviewable file.

**Review by avikganguly01 — CHANGES_REQUESTED** [review]
Looks good to me other than some files being unreviewable. I would like to see how the APIs are being used. Can you point me to your community-app PR?

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/service/SavingsApplicationProcessWritePlatformServiceJpaRepositoryImpl.java (hunk 2 above)
Comment (nikpawar89) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/service/SavingsApplicationProcessWritePlatformServiceJpaRepositoryImpl.java:98]: this has been due to squash I performed to merge several commits, I will get it corrected

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/service/SavingsAccountWritePlatformServiceJpaRepositoryImpl.java (hunk 3 above)
Comment (nikpawar89) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/service/SavingsAccountWritePlatformServiceJpaRepositoryImpl.java:369]: noted

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/DepositsApiConstants.java (hunk 4 above)
Comment (nikpawar89) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/DepositsApiConstants.java:28]: will get it corrected

**Comment by nikpawar89** [comment]
For community app,  I will have to prepare PR, by getting out gsim and gsim code.. will try get it out soon. Meanwhile we can work on getting this merged

**Comment by ippezshelby** [comment]
@edcable,  @avikganguly01 and @nikpawar89 , How soon can this be merged into Fineract including community-app side. It seems we have a need for this, please advice

**Comment by francisguchie** [comment]
@nikpawar89 , @avikganguly01 ,
How soon will this be merged into Fineract and the comm-app. There are some good updates inside of 18.03.01 that would be nice to have with this included onto them

**Comment by vorburger** [comment]
@nikpawar89 thank you very much for your interest in and contribution to Fineract/Mifos!  I'm an (old) volunteer who got reminded of Mifos/Fineract this week-end at the FOSDEM conference, and I'm taking a moment to help the project clean up some of it's old issues and pull requests etc.  

We were wondering whether you are still alive, and would be willing to `git rebase develop` this PR and take any previous review feedback into account?  If we don't hear from you within 2 weeks, we will take the liberty to close this PR without merging it, to avoid confusion for future new contributors looking at this repository - hope you understand and that's fair.

@francisguchie @Ippezrobert this needs someone to rebase and resolve conflicts, if you could do that, or find someone who could, then I'm sure other committers (probably not myself) would gladly merge this.

/Cc FYI @apache/fineract-committers

**Comment by vorburger** [comment]
Friendly service reminder: I will close this PR in 3 days to help this project have a clean list of active PRs.

**Comment by francisguchie** [comment]
@vorburger , Me and @nikpawar have been discussing about this today. He is still alive and will be responding soon on this.

**Comment by nikpawar89** [comment]
this PR is being handled in PR:
https://github.com/apache/fineract/pull/504

**Pull request closed without merging**